server/game/cards/23-AHaH/TheWardenOfTheWest.js

```diff
@@ -1,8 +1,11 @@
 const DrawCard = require('../../drawcard.js');
 const GameActions = require('../../GameActions');
+const EventAggregator = require('../../gamesteps/EventAggregator');
 
 class TheWardenOfTheWest extends DrawCard {
     setupCardAbilities(ability) {
+        this.addEventAggregator(new EventAggregator('onCardDiscarded', 'controller-aggregate', event => event.card.controller));
```

Comment:
The `setupCardAbilities` is called per card in the game, rather than once per card class. So with the current implementation, if there are multiple copies of the card in a single game (multiple copies in your own deck, both you and the opponent both have it, doesn't matter), it will get registered multiple times. Which currently will throw an error.

My recommendation, if we go with a system like this, is to register at a global level for the card (e.g. in the card implementation file, outside the class), or to simply add it to the list of aggregators you have elsewhere as needed. I'd probably lean towards the latter, since it shouldn't be very often that we need to add more aggregators.

Both of those would mean that we're always creating aggregates for them even when not needed, which you've pointed out is already a problem for the current aggregates.